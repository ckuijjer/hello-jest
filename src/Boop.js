import React from 'react';

const NotExportedComponent = () => <div>"i'm not exported"</div>;

const Boop = () => (
  <div>
    Boop
    <NotExportedComponent />
  </div>
);

export default Boop;
