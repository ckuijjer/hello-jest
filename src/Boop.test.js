import React from 'react';
import Boop from './Boop';
import TestRenderer from 'react-test-renderer'; // ES6

it('renders', () => {
  const output = TestRenderer.create(<Boop />).toJSON();
  expect(output).toMatchSnapshot();
});

it('renders NotExportedComponent', () => {
  const NotExportedComponent = Boop.__get__('NotExportedComponent');

  const output = TestRenderer.create(<NotExportedComponent />).toJSON();
  expect(output).toMatchSnapshot();
});
