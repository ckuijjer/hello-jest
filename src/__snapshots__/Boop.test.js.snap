// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div>
  Boop
  <div>
    "i'm not exported"
  </div>
</div>
`;

exports[`renders NotExportedComponent 1`] = `
<div>
  "i'm not exported"
</div>
`;
